inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_non_max_suppression.cpp

```diff
@@ -27,14 +27,14 @@ void dynamicToStaticNonMaxSuppression(std::shared_ptr<ngraph::Node> node) {
             staticShapeNMS->output(0), staticShapeNMS->output(2));
     auto dsrScores = std::make_shared<ngraph::vpu::op::DynamicShapeResolver>(
             staticShapeNMS->output(1), staticShapeNMS->output(2));
-    dsrIndices->set_friendly_name(nms->output(0).get_node_shared_ptr()->get_friendly_name());
-    dsrScores->set_friendly_name(nms->output(1).get_node_shared_ptr()->get_friendly_name());
+    dsrIndices->set_friendly_name(nms->output(0).get_node_shared_ptr()->get_friendly_name() + ".0");
```

Comment:
Looks strange. What if dynamic NMS is network output? Should we preserve operation name in that case?

Reply:
Actually the problem appears only when dynamic NMS is network output. It's a WA for the CNNNetwork over NGraph function representation, which adds `.i` to the layers outputs names if there are more than one output, so this just preserves this rule, and as a result, network outputs names. The reason why we haven't noticed that before, is that we didn't have networks with dynamic NMS output by far as well as layer tests were disabled due to absence of reference function (it was recently added, so I've enabled some of them).

Reply:
So original network outputs names have suffix ".0" and ".1"? At which stage of execution we lose them?

Reply:
There are no suffixes in the original NGraph function. They are added only by `CNNNetworkNgraphImpl` when we take network outputs. Before all the transformations, NMS is the network output and as it has multiple outputs `CNNNetworkNgraphImpl` returns `smth.0`, `smth.1`, etc. When we do this transformation we add some new layers for NMS outputs so after this DTS, NMS is not actually a network output. For instance, we add DSR to its first output and so then DSR becomes the network output. As DSR has one output, `CNNNetworkNgraphImpl` do nothing with its name, so it returns just `smth`, and we lose the outputs names and then we fail in `myriad_infer_request.cpp`

Reply:
> There are no suffixes in the original NGraph function.

When I asked `So original network outputs names have suffix ".0" and ".1"?` I meant what are outputs names in original network (what user actually has).

> They are added only by CNNNetworkNgraphImpl

Is it correct behavior? If there is an output operation and it has multiple outputs, so we have multiple network outputs which names must be kept as is, so no additional suffixes must be appended.

Reply:
If the user operates with NGraph through CNNNetwork (it's always true if user wants to pass it to the plugin), he will get the names with suffixes. It's actually a WA for CNNNetwork from NGraph representation, and as I know, there are plans to fix this for the next release. But for now, the only way to keep the network correct is to preserve this specific in the transformation itself.